### PR TITLE
Add clarification definition of stack

### PR DESCRIPTION
**Pull request by jromero**

### CODE REVIEW

File: content/docs/concepts/components/stack.md [hunk 1]
```diff
@@ -8,7 +8,10 @@ aliases=[
 
 ## What is a stack?
 
-A stack provides the buildpack lifecycle with build-time and run-time environments in the form of images.
+A stack is composed of two images that are intended to work together:
+
+1. The **build image** of a stack provides the _build-time_ environment in which the [lifecycle][lifecycle] (and thereby [buildpacks][buildpack]) are executed.
+2. The **run image** of a stack provides the _runtime_ environment in which the built application(s) would execute.
```
Comment (ekcasey) [inline content/docs/concepts/components/stack.md:14]:
```suggestion
2. The **run image** of a stack provides the base image from which application images are built.
```

File: content/docs/concepts/components/stack.md [hunk 2]
```diff
@@ -8,7 +8,10 @@ aliases=[
 
 ## What is a stack?
 
-A stack provides the buildpack lifecycle with build-time and run-time environments in the form of images.
+A stack is composed of two images that are intended to work together:
+
+1. The **build image** of a stack provides the _build-time_ environment in which the [lifecycle][lifecycle] (and thereby [buildpacks][buildpack]) are executed.
```
Comment (ekcasey) [inline content/docs/concepts/components/stack.md:13]:
```suggestion
1. The **build image** of a stack provides the base image from which the build environment is constructed. The build environment is the containerized environment in which the [lifecycle][lifecycle] (and thereby [buildpacks][buildpack]) are executed.
```

**Review by ekcasey — APPROVED** [review]
I thought about it too hard and then all the words became meaningless. Feel free to ignore me  if you find my suggestions more confusing rather than more illuminating.

**Comment by jromero** [comment]
> I thought about it too hard and then all the words became meaningless. Feel free to ignore me if you find my suggestions more confusing rather than more illuminating.

Too late, you are on the hook now. 😁

**Pull request merged**